### ISSUE TITLE
[Fortran] Smallest number is represented as zero

### ISSUE THREAD

**Issue by Thirumalai-Shaktivel** [lfortran/lfortran#2818]
```fortran
program expr2
real(4) :: r
r = 1.19209290E-07
end program
```
`--show-fortran`
```fortran
program expr2
implicit none
real(4) :: r
r = 0.000000
end program expr2
```

**Comment by Thirumalai-Shaktivel** [comment]
Same for the following cases
```fortran
r = 2.2204460492503131E-016
r = 1.17549435E-38
r = 2.2250738585072014E-308
```

**Comment by inkydragon** [comment]
https://github.com/lfortran/lfortran/blob/3a70fea9bb8f19f050c59516cf01f0cf6925958a/src/libasr/codegen/asr_to_fortran.cpp#L1269-L1277

Maybe we want to call [std::to_chars](https://en.cppreference.com/w/cpp/header/charconv) here.

https://github.com/lfortran/lfortran/blob/3a70fea9bb8f19f050c59516cf01f0cf6925958a/src/libasr/codegen/asr_to_fortran.cpp#L1274

**Comment by gerceboss** [comment]
Hi @inkydragon,
I am new to this repository and would like to contribute to this issue. Should I change the implementation to use `to_chars`?
If a more stable implementation is preferred, I suggest adding the following implementation:

```cpp
#include <sstream>

template <typename Type>
std::string to_str(const Type& t) {
  std::ostringstream os;
  os << t;
  return os.str();
}

**Comment by kmr-srbh** [comment]
@gerceboss sorry to disappoint you, it seems this issue is absent in the latest `main`. 

Output of `--show-fortran`:
```fortran
program expr2
implicit none
real(4) :: r
r = 1.19209290e-07
end program expr2
```

You can try to work on some other issue which interests you.

**Comment by certik** [comment]
@kmr-srbh do we have a test for this? If not, let's add it. Then close this issue.

**Comment by kmr-srbh** [comment]
This was fixed recently in #4122 and we have good testing for it.
